nightfall-deployer/contracts/Challenges.sol

```diff
@@ -357,7 +357,10 @@ contract Challenges is Stateful, Config {
         challengeAccepted(transaction.blockL2);
     }
 
-    function challengeHistoricRootBlockNumber(TransactionInfoBlock calldata transaction) external {
+    function challengeHistoricRootBlockNumber(
+        TransactionInfoBlock calldata transaction,
+        bytes32 salt
```

Comment:
Hey @Westlad, you added the `salt` parameter, but it is not being used anywhere. Is that correct?

Reply:
Yes, it's correct.  It still appears in the calldata (i.e. `msg.data`), even though it's not used in the function, and that's what we require. The commit hash is the hash of `msg.data.  Having a salt in there lets you change the value of the commit hash by changing the salt. 
If it's not there then the commit hash can only have one value.  Then a front-runner can just take your challenge commitment.  They can't actually challenge themselves but they can prevent you from doing so because the commit hash gets registered under their address.  Having a salt lets you have more attempts.  You may still get front run but now the front-runner has to front-run every challenger multiple times, which is a much harder task.

Reply:
Unfortunately the compiler doesn't realise it's actually being used and so issues a warning.

Reply:
I see @Westlad , then having the salt as a parameter makes the salt to be _automatically_ applied?

Reply:
Yes, that's it exactly.